src/platform/iotjs_module_uart-linux-general.inl.h

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2016 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#ifndef IOTJS_MODULE_UART_LINUX_GENERAL_INL_H
+#define IOTJS_MODULE_UART_LINUX_GENERAL_INL_H
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/signal.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include "module/iotjs_module_uart.h"
+
+void signal_handler_IO(int status);
+
+iotjs_jval_t jthis;
+int fd;
+
+
+#define UART_WORKER_INIT_TEMPLATE                                             \
+  iotjs_uart_reqwrap_t* req_wrap = iotjs_uart_reqwrap_from_request(work_req); \
+  iotjs_uart_reqdata_t* req_data = iotjs_uart_reqwrap_data(req_wrap);
+
+
+void OpenWorkerUart(uv_work_t* work_req) {
+  UART_WORKER_INIT_TEMPLATE;
+
+  jthis = req_data->jthis;
+  fd = open(iotjs_string_data(&req_data->path), O_RDWR | O_NOCTTY | O_NDELAY);
+
+  if (fd == -1) {
+    req_data->error = kUartErrOpen;
+  } else {
+    req_data->error = kUartErrOk;
+  }
+
+  struct sigaction saio;
+  saio.sa_handler = signal_handler_IO;
+  saio.sa_flags = 0;
+  saio.sa_restorer = NULL;
+  sigaction(SIGIO, &saio, NULL);
+
+  fcntl(fd, F_SETFL, FNDELAY);
+  fcntl(fd, F_SETOWN, getpid());
+  fcntl(fd, F_SETFL, O_ASYNC);
+
+  struct termios options;
+  tcgetattr(fd, &options);
+  options.c_cflag = B115200 | CS8 | CLOCAL | CREAD;
```

Comment:
Options are configurable. 
It is able to change options in javascript, but these are fixed in native.